lib/instrumentation/azure-functions.js

```diff
@@ -363,30 +363,55 @@ function instrument(agent) {
       isFirstRun = false;
     }
 
+    // In programming model v3 the InvocationContext includes
+    // `bindingDefinitions` and `executionContext`. In v4 the structure is a
+    // little different.
+    let bindingDefinitions = context.bindingDefinitions;
+    if (!bindingDefinitions) {
+      bindingDefinitions = [];
+      // Input bindings
+      bindingDefinitions.push({
+        name: context?.options?.trigger?.name,
+        type: context?.options?.trigger?.type,
+        direction: context?.options?.trigger?.direction,
+      });
+      // Output bindings
+      bindingDefinitions.push(context?.options?.return);
+    }
+    let executionContext = context.executionContext;
+    if (!executionContext) {
+      executionContext = {
+        functionDirectory: '',
+        functionName: context.functionName,
+      };
+    }
+
```

Comment:
Credit: this block is directly from https://github.com/elastic/apm-agent-nodejs/pull/4178
This is the bulk of the fix for supporting Azure Functions v4